kimchi/src/circuits/lookup/tables/mod.rs

```diff
@@ -1,45 +1,64 @@
-use ark_ff::{FftField, One, Zero};
+use ark_ff::{Field, One, Zero};
 use poly_commitment::PolyComm;
 use serde::{Deserialize, Serialize};
+use thiserror::Error;
 
 pub mod range_check;
 pub mod xor;
 
-//~ spec:startcode
-/// The table ID associated with the XOR lookup table.
-pub const XOR_TABLE_ID: i32 = 0;
-
-/// The range check table ID.
-pub const RANGE_CHECK_TABLE_ID: i32 = 1;
-//~ spec:endcode
-
-/// Enumerates the different 'fixed' lookup tables used by individual gates
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub enum GateLookupTable {
-    Xor,
-    RangeCheck,
-}
-
 /// A table of values that can be used for a lookup, along with the ID for the table.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
```

Comment:
Don't hash/check equality on a `Vec<Vec<F>>`. It could be large!

Reply:
It is absolutely true and I realise it, but otherwise we can't really check if there are any truly duplicate tables in our setup. In that case we would only be able to check table id collisions. I guess it's fine.